node/flatpak-node-generator.py

```diff
@@ -1580,6 +1693,12 @@ async def main() -> None:
         with provider_factory.create_module_provider(gen, special) as module_provider:
             with GeneratorProgress(packages, module_provider) as progress:
                 await progress.run()
+        with provider_factory.create_headers_provider(gen) as headers_provider:
+            if rcfile_node_headers:
+                with GeneratorProgress(rcfile_node_headers, headers_provider) as headers_progress:
```

Comment:
There are a few weird things here...

So GeneratorProgress in a lot of areas hardcodes text referring to packages, it also takes a ModuleProvider so I don't think this should actually type check. That being said, I don't think there would generally be enough header providers to justify using repurposing GeneratorProgress, maybe just a `print('something to do with headers')` or similar, since it will only be a few different cases I'd think. GeneratorProgress was largely added for npm because the generation can be so painfully slow there due to *so many metadata downloads*, lack of progress really makes it quite painful. But here, it would only ever be a few and in most cases one? so the time wouldn't be significant

Reply:
Well, should we then just download collected node-headers in a loop?